src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,77 @@
+import classNames from 'classnames';
+import { useMemo } from 'react';
+import { FilterType } from '../../types/Filter';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  filterType: string;
+  handleFilter: (arg: FilterType) => void;
+  deleteTodoCompleted: () => void;
+};
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  filterType,
+  handleFilter,
+  deleteTodoCompleted,
+
+}) => {
+  const activeTodos = useMemo(() => (
+    todos.filter(({ completed }) => !completed).length
+  ), [todos]);
+
+  return (
+    <footer className="todoapp__footer" data-cy="Footer">
+      <span className="todo-count" data-cy="todosCounter">
+        {`${activeTodos} items left`}
+      </span>
+
+      <nav className="filter" data-cy="Filter">
+        <a
+          data-cy="FilterLinkAll"
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: filterType === FilterType.All },
+          )}
+          onClick={() => handleFilter(FilterType.All)}
+        >
+          All
+        </a>
+
+        <a
+          data-cy="FilterLinkActive"
+          href="#/active"
+          className={classNames(
+            'filter__link',
+            { selected: filterType === FilterType.Active },
+          )}
+          onClick={() => handleFilter(FilterType.Active)}
+        >
+          Active
+        </a>
+        <a
+          data-cy="FilterLinkCompleted"
+          href="#/completed"
+          className={classNames(
+            'filter__link',
+            { selected: filterType === FilterType.Completed },
+          )}
+          onClick={() => handleFilter(FilterType.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      <button
+        data-cy="ClearCompletedButton"
+        type="button"
+        className="todoapp__clear-completed"
+        onClick={deleteTodoCompleted}
+      >
+        Clear completed
+      </button>
```

Comment:
hide this button when you don’t have completed todos
mate demo:
<img width="538" alt="Screenshot 2022-10-06 at 23 06 56" src="https://user-images.githubusercontent.com/47606379/194408405-530eab51-ed40-4280-adda-602ee34e874e.png">